### PR TITLE
refactor(react-grid): Refactor the GroupingPanel plugin

### PR DESCRIPTION
**Pull request by gsobolev**

### CODE REVIEW

File: packages/dx-react-grid-bootstrap3/package.json [hunk 1]
```diff
@@ -56,6 +56,7 @@
     "output": "../../shippable/testresults/dx-react-grid-bootstrap3.xml"
   },
   "devDependencies": {
+    "@devexpress/dx-react-grid": "1.0.0-alpha.2",
```
Comment (dxbykov) [inline packages/dx-react-grid-bootstrap3/package.json]: alpha.3 is released

File: packages/dx-react-grid-bootstrap3/package.json [hunk 2]
```diff
@@ -91,7 +92,6 @@
     "prop-types": "^15.5.8"
   },
   "peerDependencies": {
-    "@devexpress/dx-grid-core": "^1.0.0-alpha.3",
```
Comment (dxbykov) [inline packages/dx-react-grid-bootstrap3/package.json]: alpha.3 is right here

File: packages/dx-react-grid/src/components/group-panel-layout.test.jsx [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+  />
+);
+
+describe('GroupPanelLayout', () => {
+  /* eslint-disable no-console */
+  const consoleWarn = console.warn;
+  const consoleError = console.error;
+
+  const logToError = (...args) => {
+    throw new Error(format(...args).replace(/^Error: (?:Warning: )?/, ''));
```
Comment (dxbykov) [inline packages/dx-react-grid/src/components/group-panel-layout.test.jsx]: We already have the setupConsole testing utility for this.

Comment (SergeyAlexeev) [inline packages/dx-react-grid/src/components/group-panel-layout.test.jsx]: I've created the related [PR](https://github.com/DevExpress/devextreme-reactive/pull/140)

Comment (gsobolev) [inline packages/dx-react-grid/src/components/group-panel-layout.test.jsx]: Should I wait for it to be merged first?

File: packages/dx-react-grid-bootstrap3/package.json (hunk 1 above)
Comment (gsobolev) [inline packages/dx-react-grid-bootstrap3/package.json]: Looks like this change was introduced while merging the master branch. I'll fix it, thanks.

File: packages/dx-react-grid/src/components/group-panel-layout.test.jsx (hunk 3 above)
Comment (SergeyAlexeev) [inline packages/dx-react-grid/src/components/group-panel-layout.test.jsx]: Yes, there is a trouble with dependencies. Let's wait a bit.

**Pull request closed without merging**